blueprints/lib/1.0/blueprints/latency-gradient/policy.libsonnet

```diff
@@ -55,18 +56,15 @@ function(params) {
           component.withArithmeticCombinator(combinator.add(port.withConstantValue(c.concurrencyLinearIncrement),
                                                             port.withSignalName('SQRT_CONCURRENCY_INCREMENT'),
                                                             output=port.withSignalName('CONCURRENCY_INCREMENT_SINGLE_TICK'))),
-          component.withArithmeticCombinator(combinator.add(port.withSignalName('CONCURRENCY_INCREMENT_SINGLE_TICK'),
-                                                            port.withSignalName('CONCURRENCY_INCREMENT'),
-                                                            output=port.withSignalName('CONCURRENCY_INCREMENT_INTEGRAL'))),
-          component.withMin(
-            min.new()
-            + min.withInPorts({ inputs: [port.withSignalName('CONCURRENCY_INCREMENT_INTEGRAL'), port.withSignalName('ACCEPTED_CONCURRENCY')] })
-            + min.withOutPorts({ output: port.withSignalName('CONCURRENCY_INCREMENT_INTEGRAL_CAPPED') }),
-          ),
-          component.withFirstValid(
-            firstValid.new()
-            + firstValid.withInPorts({ inputs: [port.withSignalName('CONCURRENCY_INCREMENT_INTEGRAL_CAPPED'), port.withConstantValue(0)] })
-            + firstValid.withOutPorts({ output: port.withSignalName('CONCURRENCY_INCREMENT_NORMAL') }),
+          component.withIntegrator(
+            integrator.new()
+            + integrator.withInPorts({
+              input: port.withSignalName('CONCURRENCY_INCREMENT_SINGLE_TICK'),
+              max: port.withSignalName('MAX_EMA'),
```

Comment:
not sure what to put here as a max value... should I create constant with max float?

Reply:
Need to cap it ```2*ACCEPTED_CONCURRENCY```

The previous circuit with looped integral was doing the same

Reply:
So the max for the whole gradient and optimizer is the same - `NORMAL_CONCURRENCY_LIMIT`, got it.